proposals/nnnn-rpc-design-refactoring.md

```diff
@@ -0,0 +1,153 @@
+# RPC design refactoring
+
+* Proposal: [SDL-NNNN](nnnn-rpc-design-refactoring.md)
+* Author: [Alexander Kutsan](https://github.com/LuxoftAKutsan)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core]
+
+## Introduction
+
+This proposal is to rework design of application manager layer in order to:
+ - Reduce coupling between components
+ - Reduce responsibility of ApplicationManager class
+ - Provide plugin architecture for easy adding additional functionality and RPC groups   
+
+## Motivation
+
+SDL core project is actively implementing new features and new functionality.  
+To keep project on track and to reduce risks of regression during implementation
```

Comment:
To keep project on track and to reduce risks of regression during implementation
should become
To keep the project on track and to reduce risks of regression during implementation

Reply:
@mrapitis, done in 2c9f7a3f7bb7e6c8b26bb1e4aec779cf65733666

Reply:
I think this one may have also changed while you were working...

Reply:
@mrapitis, done in 3876ed33fa1f4465336267a69970a08fae9733c1
thanks